db/compaction/compaction_job.cc

```diff
@@ -1192,6 +1192,30 @@ void CompactionJob::ProcessKeyValueCompaction(SubcompactionState* sub_compact) {
       status = blob_file_builder->Finish();
     }
 
+#ifndef ROCKSDB_LITE
+    auto sfm =
+        static_cast<SstFileManagerImpl*>(db_options_.sst_file_manager.get());
+    if (status.ok() && sfm) {
+      // Report new blob files to SstFileManagerImpl
+      for (auto const& blob_file : sub_compact->blob_file_additions) {
```

Comment:
One more thing to keep in mind here, considering that compactions do not perform cleanup: `BlobFileBuilder` also exposes `blob_file_paths`, which contains all blob files, not just the successfully closed ones like `blob_file_additions`. We might want to use `blob_file_paths` here; that would also eliminate the need to form the paths below.